internal/output/output_result.go

```diff
@@ -0,0 +1,538 @@
+					continue
+				}
+
+				// Find the minium fix version
+				if minFixVersion == UnfixedDescription || semantic.MustParse(affectedEvent.Fixed, ecosystemPrefix).CompareStr(minFixVersion) < 0 {
+					minFixVersion = affectedEvent.Fixed
+				}
+			}
+		}
+	}
+
+	return minFixVersion
+}
+
+// getPackageFixedVersion determines the highest version that resolves the most known vulnerabilities for a package.
+func getPackageFixedVersion(ecosystem string, allVulns []OutputVulnResult) string {
+	ecosystemPrefix := models.Ecosystem(strings.Split(ecosystem, ":")[0])
+	maxFixVersion := ""
+	var vp semantic.Version
+	for _, vuln := range allVulns {
+		if vuln.FixedVersion == VersionUnsupported {
+			return UnfixedDescription
+		}
+
+		// Skip vulnerabilities without a fixed version.
+		if vuln.FixedVersion == UnfixedDescription {
+			continue
+		}
+
+		if maxFixVersion == "" {
+			maxFixVersion = vuln.FixedVersion
+			vp = semantic.MustParse(maxFixVersion, ecosystemPrefix)
+
+			continue
+		}
+
+		// Update if the current vulnerability's fixed version is higher
+		if vp.CompareStr(vuln.FixedVersion) < 0 {
+			maxFixVersion = vuln.FixedVersion
+			vp = semantic.MustParse(maxFixVersion, ecosystemPrefix)
+		}
+	}
+
+	// Default to UnfixedDescription if no fix version is found.
+	if maxFixVersion == "" {
+		maxFixVersion = UnfixedDescription
+	}
+
+	return maxFixVersion
+}
+
+func updateVulnCount(original OutputVulnCount, newAdded OutputVulnCount) OutputVulnCount {
+	original.SeverityCount = addSeverityCount(original.SeverityCount, newAdded.SeverityCount)
+	original.CallAnalysisCount = addCallAnalysisCount(original.CallAnalysisCount, newAdded.CallAnalysisCount)
+	original.FixableCount = addFixableCount(original.FixableCount, newAdded.FixableCount)
+
+	return original
+}
+
+func addSeverityCount(original OutputSeverityCount, newAdded OutputSeverityCount) OutputSeverityCount {
+	original.Critical += newAdded.Critical
+	original.High += newAdded.High
+	original.Medium += newAdded.Medium
+	original.Low += newAdded.Low
+	original.Unknown += newAdded.Unknown
+
+	return original
+}
+
+func addCallAnalysisCount(original OutputCallAnalysisCount, newAdded OutputCallAnalysisCount) OutputCallAnalysisCount {
+	original.Called += newAdded.Called
+	original.Uncalled += newAdded.Uncalled
+
+	return original
+}
+
+func addFixableCount(original OutputFixableCount, newAdded OutputFixableCount) OutputFixableCount {
+	original.Fixed += newAdded.Fixed
+	original.UnFixed += newAdded.UnFixed
+
+	return original
+}
+
+func increaseSeverityCount(severityCount OutputSeverityCount, typeName string) OutputSeverityCount {
+	switch typeName {
+	case "CRITICAL":
+		severityCount.Critical += 1
+	case "HIGH":
+		severityCount.High += 1
+	case "MEDIUM":
+		severityCount.Medium += 1
+	case "LOW":
+		severityCount.Low += 1
+	case "UNKNOWN":
+		severityCount.Unknown += 1
+	}
+
+	return severityCount
+}
+
+func isOSEcosystem(ecosystem string) bool {
+	for _, image := range baseImages {
```

Comment:
Where is the baseImages variable defined? Can we move it into this file if it is more appropriate.